source/Ship.cpp

```diff
@@ -4177,7 +4177,11 @@ void Ship::DoGeneration()
 		if(currentSystem)
 		{
 			double scale = .2 + 1.8 / (.001 * position.Length() + 1);
-			fuel += currentSystem->RamscoopFuel(attributes.Get("ramscoop"), scale);
+			fuel += currentSystem->RamscoopFuel(attributes.Get("ramscoop"), 1., scale);
+			fuel += currentSystem->RamscoopFuel(attributes.Get("velocity ramscoop"),
+				velocity.Length() * !IsHyperspacing() / 1.66, scale);
```

Comment:
This ends up adding the `addend` twice.
You should only be making one call to `System::RamscoopFuel`, and it should not care about the speed.
```suggestion
			double totalRamscoop = attributes.Get("ramscoop");
			totalRamscoop += IsHyperspacing() ? 0. : (attributes.Get("velocity ramscoop") * velocity.Length() / 1.66);
			fuel += currentSystem->RamscoopFuel(totalRamscoop, scale);
```
And remove the changes to `System::RamscoopFuel`.